File: golang/1493.go
func longestSubarray(nums []int) int {
	left := 0
	right := 0
	zeroCount := 0
	length := len(nums)
	longest := 0

	for right < length {
		if nums[right] == 0 {
			zeroCount++
		}
		for zeroCount > 1 {
			if nums[left] == 0 {
				zeroCount--
			}
			left++
		}
		longest = max(longest, right-left)
		right++
	}
	return longest
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: rust/1493.rs
impl Solution {
    pub fn longest_subarray(nums: Vec<i32>) -> i32 {
        let mut left = 0;
        let mut right = 0;
        let length = nums.len();
        let mut k = 1;

        while right < length {
            if nums[right] == 0 {
                k -= 1;
            }
            if k < 0 {
                if nums[left] == 0 {
                    k += 1;
                }
                left += 1;
            }
            right += 1;
        }
        (right - left - 1) as i32
    }
}
